### PR TITLE
22380-DNU-create-method-for-nil-should-not-raise-error-but-create-a-method

### PR DESCRIPTION
**Pull request by MarcusDenker**
22380 DNU "create method" for nil should not raise error but create a method
https://pharo.fogbugz.com/f/cases/22380/DNU-create-method-for-nil-should-not-raise-error-but-create-a-method

### CODE REVIEW

**Comment by MarcusDenker** [comment]
error not related

File: src/DebuggerActions/DoesNotUnderstandDebugAction.class.st [hunk 1]
```diff
@@ -95,7 +95,7 @@ DoesNotUnderstandDebugAction >> executeAction [
 	
 	msg := self interruptedContext tempAt: 1.
 	(msg lookupClass == UndefinedObject ) ifTrue: [ 
-		^self createMissingClassIn: self interruptedContext ].
+		self createMissingClassIn: self interruptedContext ].
```
Comment (macta) [inline src/DebuggerActions/DoesNotUnderstandDebugAction.class.st:98]: If we don't return here - if you have a missing class and create it won't it then continue and then try to create a method? I think I've understood that if create class fails then you assume its the nil method case - but we need to stop execution if is succeeds... so maybe something like

(self createMissingClassIn: ...) ifTrue: [ ^self ]

**Comment by macta** [comment]
I left a comment in the code - how do I download this to try it out for real? If its merged is it just in the latest build?

**Comment by MarcusDenker** [comment]
yes, it is in the latest build

**Comment by MarcusDenker** [comment]
The missing return is not a problem as the method never returns if it creates the class (due to calling 	"previousContext restart")

**Comment by macta** [comment]
Ah yes - its subtle that...

**Pull request merged**